### ISSUE TITLE
read.rsk() missing data columns

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#973]
Below is for a test file that Clark and I both have.  (I thought we had loads of rsk files in our dropbox ... ??) When I look into the rsk database I see temperature and pressure though, and neither is showing up here.

PS to Clark -- I need some more rsk files to read so I can test `as.ctd(rsk)`, which I've worked on within the `kelley` branch.

```R
> d <- read.rsk("~/Dropbox/oce-working-notes/rsk/050046_20111007_1413.rsk")
Warning message:
In read.rsk("~/Dropbox/oce-working-notes/rsk/050046_20111007_1413.rsk") :
  non-standard pressureAtmospheric value: 10.1324996948242

> summary(d)
rsk summary
-----------
* Instrument:         model RBRduo serial number 50046
* Atmosph. pressure:  10.1324996948242
* Pressure type:      absolute
* Source:             ``~/Dropbox/oce-working-notes/rsk/050046_20111007_1413.rsk``
* Time ranges from 2011-10-07 17:57:00 to 2011-10-07 18:14:02 with 1023 samples and mean step 1 s
* Statistics of data
```
          Min.                Mean                Max.                Dim. OriginalName
     time 2011-10-07 14:57:00 2011-10-07 15:05:31 2011-10-07 15:14:02 1023 -           
```
* Processing Log
    - 2016-05-13 19:20:00 UTC: ``create 'rsk' object``
    - 2016-05-13 19:20:00 UTC: ``read.rsk(file = "~/Dropbox/oce-working-notes/rsk/050046_20111007_1413.rsk")``
```

**Comment by dankelley** [comment]
I see the problem, in `R/rsk.R:818`, viz
```R
        names <- names[isMeasured] # only take names of things that are in the data table
```
the problem is that `isMeasured` is coming out to a zero-length vector in this case.

I need help from @richardsc on this, though.  Below is from my browser session. So, as can be seen, `isMeasured` is not defined in this particular database. Clark, should we be checking a version number to see if this thing will be defined? Or maybe should we simply allow *either* `isMeasured` or `isDerived`?  Or maybe this data file is so old we don't want to support it?

I'm happy to code in changes, and I think I should be the one to do it, so my eyes "get into the code" so that I can set up some more build tests relating to various rsk issues we've had lately. But I need just a bit of advice on what goes in the database.  Also, it would be terrific to have sample files in our dropbox.

I'll look at things again Sat morning. This will be the last weekend before a hoped-for release countdown so it's good that it will be poorish weather out on the grey Atlantic.

**Comment by richardsc** [comment]
Ok, this is one of those places where things get a little complicated. You're right, for that file `isMeasured` is not defined in the database. This is (apparently) just because of the age of the rsk file -- at the time it was created the schema did not contain the `isMeasured` field. It does have `isDerived`, which to my knowledge should be the opposite of `isMeasured`.

The RBR solution to database inconsistencies with old RSK files is to simply open them using the RBR software, [Ruskin](https://rbr-global.com/support/software). If you open that file, it opens directly to a screen warning that the RSK version is very old and that it should be upgraded. Ruskin will then update the file to the latest database scheme (see `050046_20111007_1413_upgraded.rsk`), which you'll see now works just fine for `read.rsk()` and does have the `isMeasured` field in it.

So what's the best solution? Given this case, we could probably code to look for either `isMeasured` or `isDerived` as you suggest. I'm fairly certain (but not 100% ... maybe 95%) that they will always be opposite, but then I don't know what the point of adding the redundant field was. I could ask next week, I suppose, for clarity.

I'll leave it up to you to decide. In the meantime, I'll start copying files to the Dropbox folder that can be used as test cases. There is also the file that gets called in the `create_data` script, which I assume you have on your machine somewhere.

**Comment by dankelley** [comment]
Thanks. I've rewritten the code to use either scheme, and to make a warning if the old scheme is detected.

I added a single build test, which I encourage Clark to examine (near the end of `tests/testthat/test_ctd.R`). This test is limited, and we can't test a whole lot more, because doing so would involve adding more `.rsk` files, which will bloat `oce` size unnecessarily.

In addition, I added a fair number of tests in a new issue to oce-issues, and this is probably much more significant. Again, I encourage Clark to have a look at that, since we don't really have the luxury of waiting a few months until a user happens upon a serious error. 

All the tests do is to check that the contents of `data` and `metadata` make some sense. Note that I skip checking `pressure`, and for that reason I'd **really** love it for Clark to look at `rsk2ctd()` in `R/rsk.R` to see that what I've done with pressure makes sense.

I am closing this issue, because I use the "open issue" list to direct coding efforts. I think we should have a new issue for anything being wrong about the columns. But, as stated, this has been addressed in `develop` commit 848fbbe7586bde6ab2c51d23280742a318362454
